server/session/player.go

```diff
@@ -67,11 +82,21 @@ const (
 	containerHotbar         = 27
 	containerInventory      = 28
 	containerOffHand        = 33
+	containerCraftingOffset = 46
 	containerBarrel         = 57
 	containerCursor         = 58
 	containerCreativeOutput = 59
+	containerCraftingResult = containerCraftingGrid + containerCraftingOffset
 )
 
+// fixID fixes the container ID passed, as it can sometimes be incorrectly sent by Minecraft. (for example, in the recipe book.)
+func fixID(id byte) byte {
```

Comment:
What's the point of this function? It's not actually incorrect, there's just different IDs for different parts of the inventory. Either way, the `invByID` function takes these into account. That should work out of the box.

Reply:
Removed.